core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToOrcTransformer.java

```diff
@@ -0,0 +1,111 @@
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.hydrator.common.HiveSchemaConverter;
+import co.cask.hydrator.common.RecordConverter;
+import org.apache.hadoop.io.BooleanWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.DoubleWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.WritableComparable;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.mapred.OrcStruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * Creates ORCStruct records from StructuredRecords
+ */
+public class StructuredToOrcTransformer extends RecordConverter<StructuredRecord, OrcStruct> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(StructuredToOrcTransformer.class);
+
+  @Override
+  public OrcStruct transform(StructuredRecord input) {
+
+    List<Schema.Field> fields = input.getSchema().getFields();
+    OrcStruct pair = parseOrcSchema(input.getSchema());
+    //populate ORC struct Pair object
+    for (int i = 0; i < fields.size(); i++) {
+      Schema.Field field = fields.get(i);
+      try {
+        WritableComparable writable = convertToWritable(field, input);
+        pair.setFieldValue(fields.get(i).getName(), writable);
+      } catch (UnsupportedTypeException e) {
+        LOG.debug(field.getName() + "is not a supported type", e);
+      }
+    }
+    return pair;
+  }
+
+  public OrcStruct parseOrcSchema(Schema inputSchema) {
+    StringBuilder builder = new StringBuilder();
+    try {
+      HiveSchemaConverter.appendType(builder, inputSchema);
+    } catch (UnsupportedTypeException e) {
+      LOG.debug("Not a valid Schema {}", inputSchema.toString(), e);
+    }
+    TypeDescription schema = TypeDescription.fromString(builder.toString());
+    OrcStruct pair = (OrcStruct) OrcStruct.createValue(schema);
+    return pair;
+  }
+
+  private WritableComparable convertToWritable(Schema.Field field, StructuredRecord input)
+    throws UnsupportedTypeException {
+    WritableComparable writable = null;
+    Object fieldVal = input.get(field.getName());
+    switch (field.getSchema().getType()) {
+      case NULL:
+        break;
+      case STRING:
+      case ENUM:
+        writable = new Text((String) fieldVal);
+        break;
+      case BOOLEAN:
+        writable = new BooleanWritable((Boolean) fieldVal);
+        break;
+      case INT:
+        writable = new IntWritable(Integer.valueOf((Integer) fieldVal));
+        break;
+      case LONG:
+        writable = new LongWritable(Long.valueOf((Long) fieldVal));
+        break;
+      case FLOAT:
+        writable = new FloatWritable((Float) fieldVal);
+        break;
+      case DOUBLE:
+        writable = new DoubleWritable(Double.valueOf((Double) fieldVal));
```

Comment:
The call to `valueOf` is unnecessary. Just cast the `fieldVal` to `Double` should be enough as the compiler will do the auto unboxing.